bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/CommunicationManager.java

```diff
@@ -470,6 +514,27 @@ public void unsetItemStateConverter(ItemStateConverter itemStateConverter) {
         this.itemStateConverter = null;
     }
 
+    @Reference(cardinality = ReferenceCardinality.AT_LEAST_ONE, policy = ReferencePolicy.DYNAMIC)
+    protected void addItemFactory(ItemFactory itemFactory) {
+        for (String itemTypeName : itemFactory.getSupportedItemTypes()) {
+            Item item = itemFactory.createItem(itemTypeName, "tmp");
+            if (item != null) {
+                acceptedCommandTypeMap.put(itemTypeName, item.getAcceptedCommandTypes());
+                acceptedStateTypeMap.put(itemTypeName, item.getAcceptedDataTypes());
```

Comment:
Hm, you are using cardinality "at least one".
Your code in remove item factory removes the item type name from the respective maps.
So, the logic in removed acts as a item type name exists in one item factory only.
If two item factories use the same item type name, it exists only once in the map and is removed if one item factory is removed.
A simple solution would be to maintain a collection with all injected item factories and rebuild the maps all the time a factory is added and removed.

Reply:
> So, the logic in removed acts as a item type name exists in one item factory only.

Yes, this was my assumption. I doubt though that there will be any deterministic outcome if multiple factories handle the same type. In any case you are right, it is safer that way.